services/s3/src/main/java/software/amazon/awssdk/services/s3/internal/UploadPartCopyRequestProvider.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.services.s3.internal;
+
+import software.amazon.awssdk.annotations.SdkInternalApi;
+import software.amazon.awssdk.services.s3.model.CopyObjectRequest;
+import software.amazon.awssdk.services.s3.model.UploadPartCopyRequest;
+
+/**
+ * Factory class to create UploadPartCopyRequest
+ */
+@SdkInternalApi
+public final class UploadPartCopyRequestProvider {
```

Comment:
Do we follow this pattern elsewhere? Any reason to not just return a collection?

Reply:
Nope, I think I just copied what we have in v1 (not the best idea 😅 ) Will refactor it.